test/TestRegistry.spec.ts

```diff
@@ -0,0 +1,390 @@
+
+      tx = await registry.delegateAvatar(delegatee3, { from: delegator });
+      expectEvent(tx, "Delegate", { delegator: delegator, avatar: avatar, delegatee: delegatee3 });
+      expect(await registry.delegate(avatar, delegatee3)).to.be.equal(true);
+
+      expect(await registry.delegate(avatar, delegatee1)).to.be.equal(true);
+      expect(await registry.delegate(avatar, delegatee2)).to.be.equal(true);
+      expect(await registry.delegate(avatar, delegatee3)).to.be.equal(true);
+    });
+
+    it("should allow a denegatee to receive multiple avatar delegation rights", async () => {
+      const delegatee = a.dummy4;
+      const delegator1 = a.user1;
+      const delegator2 = a.user2;
+      const delegator3 = a.user3;
+
+      const avatar1 = await registry.avatarOf(delegator1);
+      const avatar2 = await registry.avatarOf(delegator2);
+      const avatar3 = await registry.avatarOf(delegator3);
+
+      expect(await registry.delegate(avatar1, delegatee)).to.be.equal(false);
+      expect(await registry.delegate(avatar2, delegatee)).to.be.equal(false);
+      expect(await registry.delegate(avatar3, delegatee)).to.be.equal(false);
+
+      let tx = await registry.delegateAvatar(delegatee, { from: delegator1 });
+      expectEvent(tx, "Delegate", { delegator: delegator1, avatar: avatar1, delegatee: delegatee });
+      expect(await registry.delegate(avatar1, delegatee)).to.be.equal(true);
+
+      tx = await registry.delegateAvatar(delegatee, { from: delegator2 });
+      expectEvent(tx, "Delegate", { delegator: delegator2, avatar: avatar2, delegatee: delegatee });
+      expect(await registry.delegate(avatar2, delegatee)).to.be.equal(true);
+
+      tx = await registry.delegateAvatar(delegatee, { from: delegator3 });
+      expectEvent(tx, "Delegate", { delegator: delegator3, avatar: avatar3, delegatee: delegatee });
+      expect(await registry.delegate(avatar3, delegatee)).to.be.equal(true);
+
+      expect(await registry.delegate(avatar1, delegatee)).to.be.equal(true);
+      expect(await registry.delegate(avatar2, delegatee)).to.be.equal(true);
+      expect(await registry.delegate(avatar3, delegatee)).to.be.equal(true);
+    });
+  });
+
+  describe("revokeDelegateAvatar()", async () => {
+    it("should allow a delegator to revoke delegation rights from a delegatee", async () => {
+      const delegator = a.user1;
+      const delegatee = a.dummy1;
+      const avatar = await registry.avatarOf(delegator);
+
+      expect(await registry.delegate(avatar, delegatee)).to.be.equal(true);
+
+      const tx = await registry.revokeDelegateAvatar(delegatee, { from: delegator });
+      expectEvent(tx, "RevokeDelegate", {
+        delegator: delegator,
+        avatar: avatar,
+        delegatee: delegatee,
+      });
+
+      expect(await registry.delegate(avatar, delegatee)).to.be.equal(false);
+    });
+
+    it("should fail when avatar does not exists for the delegator", async () => {
```

Comment:
looks like the same test as in line 226

Reply:
one test is for `revokeDelegateAvatar` and another one is for `delegateAvatar`